wdl-parsing/WDLParsingFunction/src/main/java/io/dockstore/wdlparser/App.java

```diff
@@ -0,0 +1,162 @@
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.openapitools.client.model.LanguageParsingRequest;
+import org.openapitools.client.model.LanguageParsingResponse;
+import org.openapitools.client.model.VersionTypeValidation;
+import scala.Option;
+import scala.collection.JavaConverters;
+import womtool.WomtoolMain;
+
+/** Handler for requests to Lambda function. */
+public class App
+    implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+  ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public APIGatewayProxyResponseEvent handleRequest(
+      final APIGatewayProxyRequestEvent input, final Context context) {
+
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Content-Type", "application/json");
+    headers.put("X-Custom-Header", "application/json");
+
+    APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent().withHeaders(headers);
+    if (input != null && input.getBody() != null) {
+      try {
+        LanguageParsingRequest request =
+            mapper.readValue(input.getBody(), LanguageParsingRequest.class);
+        try {
+          String s =
+              parseWdlFile(
+                  request.getUri(),
+                  request.getBranch(),
+                  request.getDescriptorRelativePathInGit(),
+                  request);
+          return response.withStatusCode(HttpURLConnection.HTTP_OK).withBody(s);
+        } catch (IOException e) {
+          e.printStackTrace();
+          return response
+              .withBody("Could not clone repository to temporary directory")
+              .withStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+        } catch (GitAPIException e) {
+          StringWriter sw = new StringWriter();
+          e.printStackTrace(new PrintWriter(sw));
+          String exceptionAsString = sw.toString();
+          return response
+              .withBody(exceptionAsString)
+              .withStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
+        return response
+            .withBody("Could not process request")
+            .withStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+      }
+    } else {
+      return response
+          .withBody("No body in request")
+          .withStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+    }
+  }
+
+  private String parseWdlFile(
+      String uri,
+      String branch,
+      String descriptorRelativePathInGit,
+      LanguageParsingRequest languageParsingRequest)
+      throws IOException, GitAPIException {
+    Path tempDirWithPrefix = Files.createTempDirectory("clonedRepository");
```

Comment:
Does this need to cleaned up? Probably not given that it's a lambda, but the same instance is used again and again. I don't know what the best practices are.

Also, I'm wondering if the name shouldn't reflect the repo being cloned. Just in case something goes wrong, we might see a meaningful name in the lambda logs, so that we know we're getting an error for the foo/worklow git repo. OTOH, we may already know that from somewhere else.

Reply:
The instance is not always used again and again. Also, the temp directory (name) it creates seems to be unique every time. The lambda request already has all the important information, the tmp directory name doesn't yield anything useful